### PR TITLE
Adds support for payment transfers and reversals

### PR DESCRIPTION
**Pull request by amitmahbubani**
Adds support for payment transfers and reversals

**[Added]**
 - Create a transfer from a payment
 - List all transfers from a payment
 - Fetch all reversals for a transfer

### CODE REVIEW

**Comment by captn3m0** [comment]
@amitmahbubani please rebase against latest master

File: src/Payment.php [hunk 1]
```diff
@@ -39,6 +39,13 @@ public function capture($attributes = array())
         return $this->request('POST', $relativeUrl, $attributes);
     }
 
+    public function transfer($attributes = array())
+    {
+        $relativeUrl = $this->getEntityUrl() . $this->id . '/transfer';
```
Comment (amitmahbubani) [inline src/Payment.php]: `/transfers`

File: src/Transfer.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    public function create($attributes = array())
+    {
+        return parent::create($attributes);
+    }
+
+    /**
+     * Create a transfer to a linked account for a payment
+     */
+    public function transfer($paymentId, $attributes = array())
```
Comment (amitmahbubani) [inline src/Transfer.php]: should this function be here?

File: src/Api.php [hunk 3]
```diff
@@ -26,11 +26,13 @@ public function __construct($key, $secret)
     {
         self::$key = $key;
         self::$secret = $secret;
+
+        // Add the version to all HTTP Requests
+        Request::addHeader('User-Agent', "Razorpay-PHP/" . self::VERSION);
```
Comment (ccrims0n) [inline src/Api.php]: @albingeorge This is already happening in `src/Request.phpL147`

**Pull request merged**